anime_downloader/extractors/gcloud.py

```diff
@@ -5,27 +5,96 @@
 from anime_downloader.sites import helpers
 import logging
 
+from requests.exceptions import HTTPError
+
 logger = logging.getLogger(__name__)
 
 class Gcloud(BaseExtractor):
     def _get_data(self):
         url = self.url
         
-        """gcloud uses the same video ID as other sites"""
-        url = url.replace('fembed.com','gcloud.live')
-        url = url.replace('feurl.com','gcloud.live')
+        extVer = None
+        if not hasattr(self, 'extractorVersion'):
+            setattr(self, 'extractorVersion', 1)            
+        extVer = getattr(self, 'extractorVersion')
+        logger.debug( "extVer: {}".format(extVer))
         
-        url = url.replace('gcloud.live/v/','gcloud.live/api/source/')
-        if url.find('#') != -1:url = url[:url.find('#')] 
-        url = (url[-url[::-1].find('/'):])
-        data = helpers.post(f'https://gcloud.live/api/source/{url}').json()['data']
+        fallback_qualities = None
+        if not hasattr(self, 'fallback_qualities'):
+            setattr(self, 'fallback_qualities', None)            
+        fallback_qualities = getattr(self, 'fallback_qualities')
+        logger.debug( "fallback_qualities: {}".format(fallback_qualities))
+
 
-        if data == 'Video not found or has been removed':
-            logger.warning('File not found (Most likely deleted)')
-            return {'stream_url': ''}
+        # Reinit to V1 for next Call by an another site config
+        #setattr(self, 'extractorVersion', 1)
+        
+        if extVer == 1:
+            """gcloud uses the same video ID as other sites"""
+            url = url.replace('fembed.com','gcloud.live')
+            url = url.replace('feurl.com','gcloud.live')
+            
+            url = url.replace('gcloud.live/v/','gcloud.live/api/source/')
+            if url.find('#') != -1:url = url[:url.find('#')] 
+            url = (url[-url[::-1].find('/'):])
+            data = helpers.post(f'https://gcloud.live/api/source/{url}').json()['data']
+    
+            if data == 'Video not found or has been removed':
+                logger.warning('File not found (Most likely deleted)')
+                return {'stream_url': ''}
+            
+            for a in data:
+                if a['label'] == self.quality:
+                    return {'stream_url': a['file']}
         
-        for a in data:
-            if a['label'] == self.quality:
-                return {'stream_url': a['file']}
+        else:
+            if url.find('#') != -1:url = url[:url.find('#')]
+            urlapi = url.replace('/v/', '/api/source/' )
+            
+            data = None
+            try:
+                data = helpers.post(urlapi, referer=url).json()
+            except HTTPError as err:
+                logger.debug("_get_data HTTP Err: %s", str(err))
```

Comment:
This is redundant. The error should be raised here, as the download cannot proceed. The try except should be removed.

Reply:
Hi, ok but if the video is deleted form the server and if i have an alternative source on the _get_sources list, that can be done without this catch ?